manager/controllers/app/m4dapplication_controller.go

```diff
@@ -274,9 +287,51 @@ func (r *M4DApplicationReconciler) reconcile(applicationContext *app.M4DApplicat
 	return ctrl.Result{}, nil
 }
 
-// ConstructDataInfo gets a list of governance actions and data location details for the given dataset and fills the received DataInfo structure
+// ConstructIngestDataInfo already has the information about the external data set's endpoint,
+// but it adds the governance decisions. It also already has the link to the credentials, which was provided by the user.
+// NOTE: Assumes the credentials were put into the M4D credential manager by the user!!
+func (r *M4DApplicationReconciler) ConstructIngestDataInfo(externalDataset app.ExternalDataContext, req *modules.DataInfo, input *app.M4DApplication, clusters []m.Cluster) error {
+	dataDetails := pb.DatasetDetails{
+		DataOwner:  "",
+		DataStore:  externalDataset.ExternalStore.Connection,
+		DataFormat: externalDataset.ExternalStore.Format,
+		Geo:        string(externalDataset.Residency),
+	}
+	req.AssetID = externalDataset.ExternalStore.Connection.Name // Not cataloged so no id. Name isn't unique though.  OK?
+	req.DataDetails = &dataDetails
+	req.Credentials = nil
+	req.Actions = make(map[app.ModuleFlow]modules.Transformations)
+	req.AppInterface = &externalDataset.IFdetails
+	req.Flow = app.Copy // This is how we know it's an ingest flow
+
+	// Check if we are allowed to write the data to any of the geographies available to us
+	// If yes, we will allocated the storage later
+	excludedGeos := ""
+	for _, cluster := range clusters {
+		req.Geo = cluster.Metadata.Region
+		if err := TempLookupPolicyDecision(externalDataset, r.PolicyCompiler, req, input, pb.AccessOperation_COPY); err != nil { // TODO: Change to regular policy compiler function
```

Comment:
We need to split the lookup into different actions - one for read/copy an existing asset, another - for writing a non-cataloged asset. The API is entirely different: dataset decisions vs. general decisions. 
Not sure whether the API for writing data holds enough inputs (data residency? sensitivity?) But this is another issue.

Reply:
Agreed.  This is a temporary place holder until we do that.